doc/shared/sunadjoint/SUNAdjointStepper.rst

```diff
@@ -0,0 +1,247 @@
+
+      Jacobian-times-vector function pointer to evaluate :math:`(df/dp)^T v`.
+
+   .. c:member:: SUNJacTimesFn vJpFn
+
+      Jacobian-times-vector function pointer to evaluate :math:`v^T(df/dy)`.
+
+   .. c:member:: SUNJacTimesFn vJPpFn
+
+      Jacobian-times-vector function pointer to evaluate :math:`v^T(df/dp)`.
+
+   .. c:member:: int64_t nst
+
+      Holds the count of the number of backwards steps taken.
+
+   .. c:member:: int64_t njeval
+
+      Holds the count of the number of :math:`df/dy` evaluations.
+
+   .. c:member:: int64_t njpeval
+
+      Holds the count of the number of :math:`df/dp` evaluations.
+
+   .. c:member:: int64_t njtimesv
+
+      Holds the count of the number of :math:`(df/dy)^T v` evaluations.
+
+   .. c:member:: int64_t njptimesv
+
+      Holds the count of the number of :math:`(df/dp)^T v` evaluations.
+
+   .. c:member:: int64_t nvtimesj
+
+      Holds the count of the number of :math:`v^T(df/dy)` evaluations.
+
+   .. c:member:: int64_t nvtimesjp
+
+      Holds the count of the number of :math:`v^T(df/dp)` evaluations.
+
+   .. c:member:: int64_t nrecompute
+
+   .. c:member:: void* user_data
+
+      A pointer that is passed back to user-supplied functions
+
+   .. c:member:: void* content
+
+      Pointer to derived class specific member data
+
+   .. c:member:: SUNContext sunctx
+
+      The SUNDIALS simulation context
+
+
+The :c:type:`SUNAdjointStepper` class has the following functions:
+
+.. c:function:: SUNErrCode SUNAdjointStepper_Create(SUNStepper fwd_sunstepper, SUNStepper adj_sunstepper, \
+   int64_t final_step_idx, N_Vector sf, sunrealtype tf, SUNAdjointCheckpointScheme checkpoint_scheme, \
+   SUNContext sunctx, SUNAdjointStepper* adj_stepper)
+
+
+
+.. c:function:: SUNErrCode SUNAdjointStepper_ReInit(SUNAdjointStepper adj, N_Vector sf, sunrealtype tf)
+
+   Reinitializes the adjoint stepper to solve a new problem of the same size.
+
+   :param adj_stepper: The adjoint solver object.
+   :param tf: The time to start integrating the adjoint system from.
+   :param sf: The terminal condition vector of sensitivity solutions dg/dy0 and dg/dp.
+
+   :return: A :c:type:`SUNErrCode` indicating failure or success.
+
+
+.. c:function:: SUNErrCode SUNAdjointStepper_Evolve(SUNAdjointStepper adj_stepper, sunrealtype tout,\
+   N_Vector sens, sunrealtype* tret, int* stop_reason)
+
+   Integrates the adjoint system.
+
+   :param adj_stepper: The adjoint solver object.
+   :param tout: The time at which the adjoint solution is desired.
+   :param sens: The vector of sensitivity solutions dg/dy0 and dg/dp.
+   :param tret: On return, the time reached by the adjoint solver.
+   :param stop_reason: On return, an integer code that indicates why the adjoint solver stopped.
+
+   :return: A :c:type:`SUNErrCode` indicating failure or success.
+
+
+.. c:function:: SUNErrCode SUNAdjointStepper_OneStep(SUNAdjointStepper adj_stepper, sunrealtype tout,\
+   N_Vector sens, sunrealtype* tret, int* stop_reason)
```

Comment:
I can update this to use `last_flag` like we do with `SUNStepper` now. I had used to `stop_reason` here for consistency, but since we changed `SUNStepper` we will need to update this too.